torch_geometric/datasets/explainer_dataset.py

```diff
@@ -0,0 +1,25 @@
+from typing import Callable, Optional
+
+from torch_geometric.data import InMemoryDataset
+
+
+class ExplainerDataset(InMemoryDataset):
+    r"""Generate a custom synthetic dataset from Graph Generators.
+
+    Args:
+        generator (GraphGenerator): The graph generator instance object
+            to be used: :obj:`torch.geometric.datasets.generators.BAGraph`,
+            :obj:`torch.geometric.datasets.generators.ERGraph`.
+        transform (callable, optional): A function/transform that takes in an
+            :obj:`torch_geometric.data.Data` object and returns a transformed
+            version. The data object will be transformed before every access.
+            (default: :obj:`None`)
+    """
+    def __init__(
+        self,
+        generator,
```

Comment:
```suggestion
        generator: GraphGenerator,
```

Reply:
For some reason, this annotation or `GraphGenerator` or `Callable` is breaking`test_config_store.py` test. I ended up removing and adding to my todo for further investigation:

```
              raise ValidationError(f"Unexpected type annotation: {type_str(ref_type)}")
E               omegaconf.errors.ValidationError: Unexpected type annotation: GraphGenerator
E                   full_key: generator
E                   object_type=ExplainerDataset
```

The annotation for `transform` doesn't raise anything on this particular test.

Do you have any idea why? Should I open an issue?

Reply:
I see, then just leave it as TODO.